spec/std/socket_spec.cr

```diff
@@ -205,6 +283,30 @@ describe UNIXServer do
     File.exists?(path).should be_false
   end
 
+  it "creates the socket file with path starting with @" do
+    path = "./@crystal-test-unix-sock-starting-with-at-symbol"
+
+    File.exists?(path).should be_false
+
+    UNIXServer.open(path) do
+      File.exists?(path).should be_true
+    end
+
+    File.exists?(path).should be_false
+  end
+
+  {% if flag?(:linux) %}
+    it "does not create any file for abstract server" do
+      path = "/tmp/crystal-test-unix-abstract-sock"
+
+      File.exists?(path).should be_false
+
+      UNIXServer.open('@' + path) do
+        File.exists?(path).should be_false
```

Comment:
Maybe also doublecheck that `'@' + path` doesn't exist either? Might be paranoid...